dask_sql/context.py

```diff
@@ -542,11 +550,16 @@ def explain(
             :obj:`str`: a description of the created relational algebra.
 
         """
+        dynamic_partition_pruning = dask_config.get("sql.dynamic_partition_pruning")
+        if not dask_config.get("sql.verbose_optimizer"):
```

Comment:
What am I missing here? Seems like getting the config value for `sql.verbose_optimizer` but then setting values related to `sql.dynamic_partition_pruning`?

Reply:
So when I was working on the DPP PR, I found it pretty annoying that the `c.explain()` output would contain such long `INLIST` expressions (hundreds and thousands of values), which aren't very useful for a user who maybe wouldn't even understand where all the values are coming from. With `verbose_optimizer` set to False, though, it just affects the `c.explain()` function, temporarily turning off DPP so that we don't get the long `INLIST` expressions and instead just get what the `LogicalPlan` would be if DPP was turned off. Then at the end we switch DPP back on (if it was originally set on), so that the query itself can still run with DPP.

So if DPP is on and `verbose_optimizer` is False, DPP will still run with `c.sql()`, but `c.explain()` won't contain the DPP values.

Reply:
Conceptually I like what this is aiming to accomplish, but don't really like the idea of implicitly skipping optimizations without warning the user. Feel like maybe it might make more sense to document the introduction of the verbose logical plans in the schema for `sql.dynamic_partition_pruning`, with a suggestion to disable this config for `c.explain()` if this behavior isn't desired.

Reply:
Right now the default is to run DPP but to not have it in `c.explain()`. This is because having it in `c.explain()` would introduce filters that are not specified by the user (i.e., filters that are added by DPP) which potentially makes the `LogicalPlan` more difficult for the user to interpret. So we're not actually skipping DPP, we're just not printing how it affects the `LogicalPlan` for the user to see.

I like the idea of renaming it to `sql.explain.skipped_optimizers` or `sql.dynamic_partition_pruning.verbose` to make it more clear, although I do prefer having it default to not having DPP printed in `c.explain()` (but still run DPP by default).